integration/test/validation/app_test.go

```diff
@@ -126,97 +143,102 @@ func TestSkipValidationOnNamespaceDeletion(t *testing.T) {
-	catalogCR := &v1alpha1.Catalog{
-		ObjectMeta: metav1.ObjectMeta{
-			Name:      catalog,
-			Namespace: "default",
-		},
-		Spec: v1alpha1.CatalogSpec{
-			Description: "This catalog holds Apps exclusively running on Giant Swarm control planes.",
-			LogoURL:     "/images/repo_icons/giantswarm.png",
-			Storage: v1alpha1.CatalogSpecStorage{
-				URL:  "",
-				Type: "helm",
-			},
-			Title: catalog,
-		},
+	err = createNamespace(ctx, orgNamespace)
+	if err != nil {
+		t.Fatalf("expected nil but got error %#v", err)
 	}
 
 	app := &v1alpha1.App{
 		ObjectMeta: metav1.ObjectMeta{
 			Name:      appName,
-			Namespace: namespace,
+			Namespace: orgNamespace,
 			Labels: map[string]string{
-				label.AppOperatorVersion: "3.0.0",
+				label.AppOperatorVersion: "5.5.0",
 			},
 		},
 		Spec: v1alpha1.AppSpec{
-			Catalog: catalog,
-			Config: v1alpha1.AppSpecConfig{
-				ConfigMap: v1alpha1.AppSpecConfigConfigMap{
-					Name:      configMapName,
-					Namespace: namespace,
-				},
-			},
+			Catalog:   catalog,
 			Name:      appName,
-			Namespace: namespace,
+			Namespace: "kube-system",
 			KubeConfig: v1alpha1.AppSpecKubeConfig{
 				InCluster: true,
 			},
 			Version: "1.2.2",
 		},
 	}
+	expectedError := "validation error: target namespace kube-system is not allowed for in-cluster apps"
+
+	logger.Debugf(ctx, "waiting for failed app creation")
 
 	o := func() error {
-		_, err = appTest.K8sClient().CoreV1().Namespaces().Create(ctx, ns, metav1.CreateOptions{})
-		if apierrors.IsAlreadyExists(err) {
-			// fall through
-			return nil
-		} else if err != nil {
-			return microerror.Mask(err)
+		err = appTest.CtrlClient().Create(ctx, app)
+		if err == nil {
+			return microerror.Maskf(executionFailedError, "expected error but got nil")
 		}
-
-		_, err = appTest.K8sClient().CoreV1().ConfigMaps(namespace).Create(ctx, cm, metav1.CreateOptions{})
-		if apierrors.IsAlreadyExists(err) {
-			// fall through
-			return nil
-		} else if err != nil {
-			return microerror.Mask(err)
+		if !strings.Contains(err.Error(), expectedError) {
+			return microerror.Maskf(executionFailedError, "error == %#v, want %#v ", err.Error(), expectedError)
 		}
 
-		err = appTest.CtrlClient().Create(ctx, catalogCR)
-		if apierrors.IsAlreadyExists(err) {
-			// fall through
-			return nil
-		} else if err != nil {
-			return microerror.Mask(err)
-		}
+		return nil
+	}
+	b := backoff.NewConstant(5*time.Minute, 10*time.Second)
+	n := backoff.NewNotifier(logger, ctx)
 
-		err = appTest.CtrlClient().Create(ctx, app)
-		if apierrors.IsAlreadyExists(err) {
+	err = backoff.RetryNotify(o, b, n)
+	if err != nil {
+		t.Fatalf("expected nil but got error %#v", err)
+	}
+
+	logger.Debugf(ctx, "waited for failed app creation")
+}
+
+// TestSkipValidationOnNamespaceDeletion tests that when the namespace
+// containing an app CR is deleted the validation logic is skipped. This is
+// done by checking if the app CR has a deletion timestamp.
```

Comment:
Just to add context. This changes with the org namespace. 

Now we delete the cluster namespace rather than the app CRs individually.

With the org namespace the customer is responsible for deleting the apps in the org namespace. cluster-apps-operator then deletes chart-operator and app-operator.

We need to support both types of deletion so it would be good to have coverage for that.

Reply:
Just to let you know, I've seen your comment yesterday and I will of course take care of it, but as we have just spoken, I will move these test-related changes into another PR.